src/app/products/home/home.component.ts

```diff
@@ -39,64 +33,61 @@ import { CreateproductmodalComponent } from '../createproductmodal/createproduct
     MatSnackBarModule,
     CommonModule,
     CreateproductmodalComponent,
+    ReactiveFormsModule,
+    NgOptimizedImage,
   ],
 })
 export class HomeComponent implements OnInit {
-  horizontalPosition: MatSnackBarHorizontalPosition = 'end';
-  verticalPosition: MatSnackBarVerticalPosition = 'bottom';
-
-  newProducts$!: Observable<ProductData[]>;
-
-  products!: Signal<ProductData[]>;
-  currentPage: number = 1;
-  itemsPerPage: number = 8;
-  loading!: Signal<boolean>;
-  productLength: number = 0;
-  durationInSeconds = 2;
-  toggleModal: boolean = false;
-
-  @ViewChild('cardRef', { read: ElementRef })
-  card!: ElementRef;
+  public horizontalPosition: MatSnackBarHorizontalPosition = 'end';
+  public verticalPosition: MatSnackBarVerticalPosition = 'bottom';
+  public newProducts$!: Observable<ProductData[]>;
+  public searchForm!: FormGroup;
+  public products = this.store.selectSignal(allProducts);
+  public currentPage = 1;
+  public itemsPerPage = 8;
+  public loading = this.store.selectSignal(isProductsLoading);
+  public productLength = 0;
+  public durationInSeconds = 2;
+  public toggleModal = false;
```

Comment:
You mean all this variables are public and are been use outside this component?

Reply:
Not all
I have made only the ones used outside the component public